lib/portlayer/storage/volume_cache.go

```diff
@@ -35,22 +36,68 @@ type VolumeLookupCache struct {
 	vlc     map[string]Volume
 	vlcLock sync.RWMutex
 
-	// The underlying data storage implementation
-	volumeStore VolumeStorer
+	// Maps the service url of the volume store to the underlying data storage implementation
+	volumeStores map[url.URL]VolumeStorer
```

Comment:
https://golang.org/pkg/net/url/#URL

I guess you mean UserInfo.  Good call.  I'll change this to a string.  I'm using that idiom all over the place so I'll push a new PR for the other places too after this one.